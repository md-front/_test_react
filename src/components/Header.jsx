import React from 'react';
import Form from './Form';
import styles from '../styles/components/Header.module.scss';
import {connect} from "react-redux";
import {clearList} from "../redux/actions/app";

const btns = [
    {
        type: 'favorites',
        className: 'fav',
        disableClassName: 'fav-disabled',
        text: 'Очистить избранное',
    },
    {
        type: 'blacklist',
        className: 'del',
        disableClassName: 'del-disabled',
        text: 'Отобразить скрытые вакансии',
    },
]

const Header = props => (
    <header>
        <div className="container">

            <div className={styles.actions}>
                <a href={window.location.origin + window.location.pathname}
                   className={styles.rootLink}
                   title="Применить стандартный поиск">
                    <h1 className={styles.title}>Поиск вакансий</h1>
                </a>

                <div className={styles.btns}>
                    {btns.map((btn, index) =>
                        <button type="button"
                                key={index}
                                className={props[btn.type].length ? styles[btn.className] : styles[btn.disableClassName]}
                                onClick={() => props.clearList(btn.type)}>
                            {btn.text}
                        </button>
                    )}
                </div>
            </div>

            <Form />

        </div>
    </header>
)

const mapStateToProps = ({app}) => ({
    blacklist: app.blacklist,
    favorites: app.favorites
})


const mapDispatchToProps = {
    clearList
}

export default connect(mapStateToProps, mapDispatchToProps)(Header)