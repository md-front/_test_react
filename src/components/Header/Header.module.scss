@import '/src/styles/variables.scss';

.actions {
  display: flex;
  margin: 20px 0 30px;

  @media ($mobileL) {
    display: block;
  }
}

.rootLink {
  text-decoration: none;
  color: $textColor;
  transition: color .2s;

  &:hover {
    color: $customPurple;
  }
}

.title {
  font-weight: 500;
  font-size: 32px;
}

.btns {
  display: flex;
  margin-left: auto;

  @media ($mobileL) {
    justify-content: space-between;
    margin-top: 15px;
  }
}

/* &[label] */
.showArchived {
  display: inline-flex;
  margin-left: 15px;
  align-items: center;
  font-size: 15px;
  font-weight: 500;
  cursor: pointer;

  @media ($mobileL) {
    margin: 0;
  }

  @media ($mobileP) {
    padding: 5px 8px;
    font-size: 12px;
  }

  input {
    display: none;

    /* switch wrap */
    &:checked+span {

      /* switch inner icon */
      &>span {
        transform: translateX(16px);
        background-color: $customPurpleH;
      }

      &+span {
        opacity: 1;
      }
    }
  }

  input {

    /* switch wrap */
    &+span {
      display: inline-flex;
      width: 40px;
      height: 23px;
      margin-right: 10px;
      padding: 2px;
      align-items: center;
      border: 2px solid $customPurpleL;
      border-radius: $commonBorderRadius;
      background-color: #fff;

      /* switch inner icon */
      span {
        width: 16px;
        height: 16px;
        background-color: $customPurpleL;
        border-radius: 50%;
        transition: .2s;
      }

      /* text */
      &+span {
        opacity: .7;
        transition: .2s;

        &:hover {
          opacity: 1;
        }
      }
    }
  }
}

/* &[button] */
.btn {
  display: inline-flex;
  margin-left: 15px;
  align-items: center;
  padding: 5px 10px;
  font-size: 15px;
  font-weight: 500;
  border: 2px solid;
  border-radius: $commonBorderRadius;
  background-color: transparent;
  transition: .2s;
  cursor: pointer;

  @media ($mobileL) {
    margin: 0;
  }

  @media ($mobileP) {
    padding: 5px 8px;
    font-size: 12px;
  }
}

.disabled {
  opacity: .4;
  pointer-events: none;
}

.arch {
  composes: btn;
  border-color: $customRed;

  &:hover {
    background-color: $customRed;
  }

  &-disabled {
    composes: arch;
    composes: disabled;
  }
}

.fav {
  composes: btn;
  border-color: $favColor;

  &:hover {
    background-color: $favColor;
  }

  &-disabled {
    composes: fav;
    composes: disabled;
  }
}

.del {
  composes: btn;
  border-color: $defaultColor;

  &:hover {
    background-color: $defaultColor;
  }

  &-disabled {
    composes: del;
    composes: disabled;
  }
}