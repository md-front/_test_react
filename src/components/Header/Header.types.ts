import { imprintFav } from '../../types/initialParams.types';

export interface HeaderProps {
    imprintFav: Array<imprintFav>,
    blacklist: Array<number>,
    favorites: Array<number>,
    showArchived: boolean,
    clearList: (type: keyof HeaderProps) => void
    toggleArchived: (showArchived: boolean) => void
}

export interface Btn {
    type: keyof HeaderProps,
    className: string,
    disableClassName: string,
    text: string,
    tip: string,
}

// TODO
export interface AppProps {
    app: HeaderProps
}
