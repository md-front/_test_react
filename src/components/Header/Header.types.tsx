export interface HeaderProps {
    blacklist: Array<number>,
    favorites: Array<number>,
    clearList: (type: keyof HeaderProps) => void
}

export interface Btn {
    type: keyof HeaderProps,
    className: string,
    disableClassName: string,
    text: string,
}

// TODO
export interface HeaderState {
    app: HeaderProps
}