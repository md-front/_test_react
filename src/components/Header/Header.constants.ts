import { Btn } from './Header.types';

export const BUTTONS_DATA: Array<Btn> = [
  // {
  //   type: 'imprintFav',
  //   className: 'arch',
  //   disableClassName: 'arch-disabled',
  //   text: 'Очистить архив',
  //   tip: 'Старые данные удалены',
  // },
  {
    type: 'favorites',
    className: 'fav',
    disableClassName: 'fav-disabled',
    text: 'Очистить избранное',
    tip: '',
  },
  {
    type: 'blacklist',
    className: 'del',
    disableClassName: 'del-disabled',
    text: 'Отобразить скрытые вакансии',
    tip: '',
  },
];
