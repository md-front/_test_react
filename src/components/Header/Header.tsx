import React from 'react';
import { connect } from 'react-redux';
import { Form } from '../Form';
import styles from './Header.module.scss';
import { clearList } from '../../redux/actions/app';
import { AppProps, HeaderProps } from './Header.types';
import { BUTTONS_DATA } from './Header.constants';

function Header(props: HeaderProps) {
  return (
    <header>
      <div className="container">

        <div className={styles.actions}>
          <a
            href={window.location.origin + window.location.pathname}
            className={styles.rootLink}
            title="Применить стандартный поиск"
          >
            <h1 className={styles.title}>Поиск вакансий</h1>
          </a>

          <div className={styles.btns}>
            {BUTTONS_DATA.map((btn) => (
              <button
                type="button"
                key={btn.type}
                className={
                  props[btn.type].length ? styles[btn.className] : styles[btn.disableClassName]
                }
                onClick={() => props.clearList(btn.type)}
              >
                {btn.text}
              </button>
            ))}
          </div>
        </div>

        {/* @ts-ignore TODO */}
        <Form />

      </div>
    </header>
  );
}

const mapStateToProps = ({
  app: {
    blacklist,
    favorites,
  },
}: AppProps) => ({
  blacklist,
  favorites,
});

const mapDispatchToProps = {
  clearList,
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
