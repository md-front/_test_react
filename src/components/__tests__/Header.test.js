import React from 'react';
import { render, screen } from '@testing-library/react';
import Header from '../Header';

it('Рендер', () => {
    const header = render(<Header />)
    expect(header).toMatchSnapshot();
    expect(screen.getByText('Поиск вакансий')).toBeInTheDocument();
});