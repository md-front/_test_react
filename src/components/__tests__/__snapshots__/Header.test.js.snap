// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Рендер 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="container"
        >
          <div
            class="actions"
          >
            <h1
              class="title"
            >
              Поиск вакансий
            </h1>
            <div
              class="btns"
            >
              <button
                class="fav-disabled"
                type="button"
              >
                Очистить избранное
              </button>
              <button
                class="del-disabled"
                type="button"
              >
                Отобразить скрытые вакансии
              </button>
            </div>
          </div>
          <form
            class="form"
            title="work in progress"
          >
            <div
              class="form__inner"
            >
              <input
                class="form__input"
                disabled=""
                placeholder="Frontend"
                type="text"
                value="Frontend"
              />
              <button
                class="form__btn"
                disabled=""
                type="submit"
              >
                Найти
              </button>
            </div>
            <button
              class="form__more"
              disabled=""
              type="button"
            >
              <svg>
                search-more.svg
              </svg>
            </button>
          </form>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="container"
      >
        <div
          class="actions"
        >
          <h1
            class="title"
          >
            Поиск вакансий
          </h1>
          <div
            class="btns"
          >
            <button
              class="fav-disabled"
              type="button"
            >
              Очистить избранное
            </button>
            <button
              class="del-disabled"
              type="button"
            >
              Отобразить скрытые вакансии
            </button>
          </div>
        </div>
        <form
          class="form"
          title="work in progress"
        >
          <div
            class="form__inner"
          >
            <input
              class="form__input"
              disabled=""
              placeholder="Frontend"
              type="text"
              value="Frontend"
            />
            <button
              class="form__btn"
              disabled=""
              type="submit"
            >
              Найти
            </button>
          </div>
          <button
            class="form__more"
            disabled=""
            type="button"
          >
            <svg>
              search-more.svg
            </svg>
          </button>
        </form>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Рендер default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="container"
        >
          <div
            class="actions"
          >
            <h1
              class="title"
            >
              Поиск вакансий
            </h1>
            <div
              class="btns"
            >
              <button
                class="fav-disabled"
                type="button"
              >
                Очистить избранное
              </button>
              <button
                class="del-disabled"
                type="button"
              >
                Отобразить скрытые вакансии
              </button>
            </div>
          </div>
          <form
            class="form"
            title="work in progress"
          >
            <div
              class="form__inner"
            >
              <input
                class="form__input"
                disabled=""
                placeholder="Frontend"
                type="text"
                value="Frontend"
              />
              <button
                class="form__btn"
                disabled=""
                type="submit"
              >
                Найти
              </button>
            </div>
            <button
              class="form__more"
              disabled=""
              type="button"
            >
              <svg>
                search-more.svg
              </svg>
            </button>
          </form>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="container"
      >
        <div
          class="actions"
        >
          <h1
            class="title"
          >
            Поиск вакансий
          </h1>
          <div
            class="btns"
          >
            <button
              class="fav-disabled"
              type="button"
            >
              Очистить избранное
            </button>
            <button
              class="del-disabled"
              type="button"
            >
              Отобразить скрытые вакансии
            </button>
          </div>
        </div>
        <form
          class="form"
          title="work in progress"
        >
          <div
            class="form__inner"
          >
            <input
              class="form__input"
              disabled=""
              placeholder="Frontend"
              type="text"
              value="Frontend"
            />
            <button
              class="form__btn"
              disabled=""
              type="submit"
            >
              Найти
            </button>
          </div>
          <button
            class="form__more"
            disabled=""
            type="button"
          >
            <svg>
              search-more.svg
            </svg>
          </button>
        </form>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
